source/swap/contracts/Swap.sol

```diff
@@ -361,17 +362,18 @@ contract Swap is ISwap {
 
     // Ensure the transfer is not to self.
     require(from != to, "INVALID_SELF_TRANSFER");
-
-    if (kind == ERC721_INTERFACE_ID) {
-
-      require(amount == 0, "NO_AMOUNT_FIELD_IN_ERC721");
-      // Attempt to transfer an ERC-721 token.
-      IERC721(token).transferFrom(from, to, id);
-
-    } else {
-      require(id == 0, "NO_ID_FIELD_IN_ERC20");
-      // Attempt to transfer an ERC-20 token, underlying SafeERC20 calls require.
-      IERC20(token).safeTransferFrom(from, to, amount);
-    }
+    ITransferHandler transferHandler = registry.getTransferHandler(kind);
```

Comment:
would it be easier to return `transferHandler` as an address as it is always cast to an address, and never used as an `ITransferHandler`?